### PR TITLE
AppsController split: `replace app` method

### PR DESCRIPTION
**Pull request by meln1k**
This is another part of migration to akka-http

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala [hunk 1]
```diff
@@ -222,6 +259,14 @@ class AppsController(
 
   private def getVersion(appId: PathId, version: Timestamp)(implicit identity: Identity): Route = ???
 
+  private def completeWithDeploymentForApp(appId: PathId, plan: DeploymentPlan) =
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala]: Could you add a comment so refactoring would be easier? Is this used somewhere else?

File: src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala [hunk 2]
```diff
@@ -206,7 +209,41 @@ class AppsController(
...
+    * @return http servlet response
+    */
+  private[this] def update(appId: PathId, partialUpdate: Boolean, allowCreation: Boolean)(implicit identity: Identity): Route = {
+    val version = clock.now()
+
+    (forceParameter &
+      extractClientIP &
+      extractUri &
+      entity(as(appUpdateUnmarshaller(appId, partialUpdate)))) { (force, remoteAddr, requestUri, appUpdate) =>
+        //         Note - this function throws exceptions and handles authorization synchronously. We need to catch and map these
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala]: Nit: What is the reason for the space in the comment?

File: src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala [hunk 3]
```diff
@@ -206,7 +209,41 @@ class AppsController(
...
+        //         exceptions to the appropriate rejections
+        val fn = updateOrCreate(
+          appId, _: Option[AppDefinition], appUpdate, partialUpdate, allowCreation, clock.now(), marathonSchedulerService)
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala]: Hm, this is an interesting pattern. Why not create a proper method with `def`? What is the motivation for using the underscore?

File: src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala [hunk 4]
```diff
@@ -206,7 +209,41 @@ class AppsController(
...
+
+        onSuccessLegacy(Some(appId))(groupManager.updateApp(appId, fn, version, force)).apply { plan =>
```
Comment (jeschkies) [inline src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala]: It's really hard to follow all the nested methods :/. It feels it would make more sense to factor out `groupManager.updateApp(appId, fn, version, force)).apply ...` and not `completeWithDeploymentForApp`. ...ok, I throught `apply` belong to the update but it's part of the `onComplete` directive returned by  `onSuccessLegacy`. I have a hard time accepting this. We mix directives and actual logic too much.

File: src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala (hunk 2 above)
Comment (meln1k) [reply src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala]: IntelliJ misbehaved.

File: src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala (hunk 4 above)
Comment (meln1k) [reply src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala]: I agree, but refactoring will require quite a lot of changes, let's do it in another PR.

File: src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala (hunk 1 above)
Comment (meln1k) [reply src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala]: 1. sure
2. yes, it is, but not in this PR
